src/template-instance.ts

```diff
@@ -25,7 +25,9 @@ function* collectParts(el: DocumentFragment): Generator<TemplatePart> {
         }
       }
     } else if (node instanceof Text && node.textContent && node.textContent.includes('{{')) {
-      for (const token of parse(node.textContent)) {
+      const parsed = parse(node.textContent)
+      for (let i = 0; i < parsed.length; i += 1) {
+        const token = parsed[i]
```

Comment:
Small comment: Do you think a code comment with a link to this PR for context would be helpful?

I'd be a little worried that someone sees this code and refactors it back to `for const ... in`, resulting in a regression.

Reply:
This sounds like a great idea! 

I'll merge this once a comment is added 😝